ray-curator/ray_curator/stages/deduplication/exact/exact_duplicate_identification.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Any, Literal
+
+from ray_curator.stages.deduplication.id_generator import CURATOR_DEDUP_ID_STR, get_id_generator_actor
+from ray_curator.stages.deduplication.io_utils import DeduplicationIO
+from ray_curator.stages.shuffler.rapidsmpf_shuffler import pylibcudf_to_cudf_dataframe
+from ray_curator.stages.shuffler.stage import ShuffleStage
+from ray_curator.tasks import FileGroupTask
+from ray_curator.utils.file_utils import get_fs
+
+if TYPE_CHECKING:
+    import cudf
+
+    from ray_curator.backends.base import WorkerMetadata
+
+
+EXACT_DUPLICATE_GROUP_FIELD = "_exact_duplicate_group"
+
+
+class ExactDuplicateIdentification(DeduplicationIO, ShuffleStage):
+    """
+    Stage that finds exact duplicates in a given column.
+
+    Parameters
+    ----------
+    text_field
+        Field name representing the field to find duplicates in.
+    output_path
+        Path to write output files.
+    read_kwargs
+        Keyword arguments for cudf.read_parquet method.
+    write_kwargs
+        Keyword arguments for cudf.to_parquet method.
+    total_nparts
+        Total number of output partitions. If None, will be set automatically by the executor.
+    rmm_pool_size
+        Size of the RMM GPU memory pool in bytes.
+        If "auto", the memory pool is set to 90% of the free GPU memory.
+        If None, the memory pool is set to 50% of the free GPU memory that can expand if needed.
+    spill_memory_limit
+        Device memory limit in bytes for spilling to host.
+        If "auto", the limit is set to 80% of the RMM pool size.
+        If None spilling is disabled.
+    enable_statistics
+        Whether the underlying rapidsmpf shuffler should collect shuffle statistics.
+    """
+
+    _name = "ExactDuplicateIds"
+
+    def __init__(  # noqa: PLR0913
+        self,
+        text_field: str,
+        output_path: str,
+        input_filetype: Literal["jsonl", "parquet"] = "jsonl",
```

Comment:
maybe we can add an arg to say assign_ids.. since in theory Exact might work _fine_ without id generator stuff too

Reply:
If not I would also need an `id_field` right?

Reply:
Yuppp